proto/ate.proto

```diff
@@ -643,6 +643,13 @@ message Ipv4Header {
   uint32 dscp = 4;
   uint32 ecn = 5;
   uint32 checksum = 7;
+  uint32 header_length = 8;
+  UIntRange protocol = 9;
+  uint32 total_length = 10;
+  bool identification = 11;
+  bool reserved = 12;
+  bool more_fragments = 13;
+  UIntRange fragment_offset = 14;
```

Comment:
What was the aim of using a UIntRange here? Was it (a) to distinguish between omitting the value vs someone explicitly setting it to zero; or (b) to eventually allow the user to specify a range here? Assuming it was (a), I want to confirm: that means there is a legitimate case where IxNetwork would default the value to a non-zero integer AND the user legitimately wants to explicitly set it to zero.  That could happen?

Reply:
Yes that is the correct assumption (a). IxNetwork default protocol field is <auto>. At the same time protocol value 0 (HOPOPT) is also a valid one.